### PR TITLE
Enhance and simplify logging guide

### LINKED ISSUE #540: Use SLF4J in logging guide
The following idiom is recommended in the logging guide:

```java
if (log.isEnabled(Logger.Level.TRACE)) {
    log.trace("called method with: " + name);
}
```
SLF4J supports a feature called [parameterized logging](https://slf4j.org/faq.html#logging_performance) which achieves the same result on one line: 
```java
log.trace("called method with: {}", name);
```

As [SLF4J binding to JBoss Logging](https://github.com/jboss-logging/slf4j-jboss-logging) is provided by Wildfly, I suggest that we recommend to use it in the logging guide instead to advocate clean code. [Here](https://github.com/jboss-logging/slf4j-jboss-logging/blob/main/src/main/java/org/jboss/slf4j/JBossLoggerAdapter.java#L99) is the corresponding adapter that proves that the two code snippets above are equivalent.

### PR DESCRIPTION
**Pull request by hpehl**
Fixes #540

### CODE REVIEW

**Comment by hpehl** [comment]
@jamezp, any objections to merging this?

**Review by asoldano — APPROVED** [review]

File: guides/application-logging.adoc [hunk 1]
```diff
@@ -57,10 +57,7 @@ public class GettingStartedService {
     private static Logger log = Logger.getLogger(GettingStartedService.class.getName());
 
     public String hello(String name) {
-        if (log.isEnabled(Logger.Level.TRACE)) {
-            log.trace("called method with: " + name);
-        }
-
+        log.trace("called method with: {}", name);
```
Comment (jamezp) [inline guides/application-logging.adoc:60]: This isn't quite correct. For a format like this it needs to be `tracev`. However, I'd suggest using:
```java
log.tracef("called method: %s", name);
```

**Review by jamezp — CHANGES_REQUESTED** [review]
I actually see other things I'd like to change in here :) Would you prefer I make a PR to update this in general?

**Comment by jamezp** [comment]
> @jamezp I agree it's better you open a separate PR to adjust the content.

Let me know what you think about #542 @hpehl. The plugin configuration could remove. However, it won't work out of the box the way that is currently shown. We might need a more uniform approach in general to that part of the message though.

**Comment by bstansberry** [comment]
I've put the 'hold' label on this due to the discussion around #542. The label is just to make the fact there's an ongoing 'blocking' discussion more visible when looking at the PR queue.

**Comment by hpehl** [comment]
@jamezp, I think we can close this PR now that https://github.com/wildfly/wildfly.org/pull/542 has been merged.

**Pull request closed without merging**